samples/teamsfx-app/src/components/App.tsx

```diff
@@ -0,0 +1,64 @@
+// https://fluentsite.z22.web.core.windows.net/quick-start
+import {
+  FluentProvider,
+  teamsLightTheme,
+  teamsDarkTheme,
+  teamsHighContrastTheme,
+  tokens
+} from '@fluentui/react-components';
+import { useEffect } from 'react';
+import { HashRouter as Router, Navigate, Route, Routes } from 'react-router-dom';
+import { app } from '@microsoft/teams-js';
+import { useTeamsUserCredential } from '@microsoft/teamsfx-react';
+import Privacy from './Privacy';
+import TermsOfUse from './TermsOfUse';
+import Tab from './Tab';
+import TabConfig from './TabConfig';
+import { TeamsFxContext } from './Context';
+
+/**
+ * The main app which handles the initialization and routing
+ * of the app.
+ */
+export default function App() {
+  const { loading, theme, themeString, teamsUserCredential } = useTeamsUserCredential({
+    clientId: process.env.REACT_APP_CLIENT_ID!,
```

Comment:
Maybe I'm missing it but where are these environment variable being set?

Reply:
ahh, got it, in the yml that teams toolkit is using to launch things from the variables in the .env file, so much redirection